### ISSUE TITLE
Role list limit bug for contact permission

### ISSUE THREAD

**Issue by etidimjav** [sulu/sulu#5680]
| Q | A
| --- | ---
| Sulu Version | 2.1.2

# Actual Behavior

The role list in contact permission is blocked to 10 items despite there is more than 10 roles.

# Expected Behavior

No limit for this list.

**Comment by nnatter** [comment]
Thanks for reporting! It looks like the `ResourceMultiSelect` (which is used in the `RoleAssignments`) does not pass any `limit` parameter to the `ResourceListStore`. Because of this, the API uses the default limit of 10.

**Comment by ymc-thzi** [comment]
Having the same issue. Is there some workaround?

**Comment by nnatter** [comment]
I think you could overwrite/decorate the `ListRestHelper` in your project and adjust the behaviour of the `getLimit` method: 

- https://symfony.com/doc/current/bundles/override.html#services-configuration
- https://github.com/sulu/sulu/blob/master/src/Sulu/Component/Rest/ListBuilder/ListRestHelper.php#L101-L115

**Comment by alexander-schranz** [comment]
The simpliest way would be create a [Request EventSubscriber](https://symfony.com/doc/current/event_dispatcher.html#event-aliases). Which does something like `$request->query->set('limit', '');`  like:

```php
        if (!$event->isMasterRequest()) {
            // don't do anything if it's not the master request
            return;
        }

        $request = $event->getRequest();

        if (!$request .. ) { // do nothing when not roles api is called
               return;
        }

       $request->query->set('limit', ''); // set to empty to define no limit or set to something like 9999 to get all roles
```

**Comment by danrot** [comment]
Fixed in #5707. Thanks for reporting!
